### DISCUSSION TITLE
insta360 projection

### DISCUSSION THREAD

**Discussion by cuhlik** [mistic100/Photo-Sphere-Viewer#1328] [General]
Does anyone know what projection the insta360 X3, X4, etc cameras use for their full-panoramas.  The .insp files are 8704x4352 pixel JPEG files containing two round images that look like they came directly from the image sensors behind fish-eye lenses.  I.e. not equirectangular projections.  I'd like to either convert my .insp images to equirectangular images or add code to photosphere.js to handle .insp files directly.  The attached file is a .insp file directly from the camera, but with its extension changed to .jpg so it will upload.
![IMG_20240613_153216_00_004](https://github.com/mistic100/Photo-Sphere-Viewer/assets/2542009/d0a51883-6491-43e8-a450-2e479d16149a)

**Comment by mistic100** [comment]
Please search issues before posting https://github.com/mistic100/Photo-Sphere-Viewer/issues/1288

**Comment by cuhlik** [comment]
Thank-you, thank-you, thank-you!  Works great!
I just needed a little bit of scaling to tweak the insta360 dual-fisheye
images to be closer to 180° instead of the ~190° provided by the camera.
To make the seam go away, actual image stitching magic needs to be
applied...

In case you are interested, here is an example where each of the fish-eye
halves is scaled up by 5.5% (about the best I could do with simple scaling).

https://fuz-moto.com/trips/tst_trip/tst_insta360.html

Chris
